bpf/process/types/basic.h

```diff
@@ -1337,13 +1351,17 @@ filter_read_arg(void *ctx, int index, struct bpf_map_def *heap,
 		struct bpf_map_def *config_map)
 {
 	struct msg_generic_kprobe *e;
+	struct event_config *config;
 	int pass, zero = 0;
 	size_t total;
 
 	e = map_lookup_elem(heap, &zero);
 	if (!e)
 		return 0;
-	pass = filter_args(e, index, filter);
+	config = map_lookup_elem(config_map, &e->idx);
+	if (!config)
+		return 0;
+	pass = filter_args(e, index, filter, config->flags & 1);
```

Comment:
Maybe we can do `config->flags & FLAGS_EARLY_FILTER` instead of `config->flags & 1` here in order to make it easier to understand.